### PR TITLE
[BE] Cleanup empty lines

### PR DESCRIPTION
**Pull request by XuehaiPan**
Clean up empty lines and always have a newline at the end-of-file. The leading and tailing lines are trimed by:

```bash
fd --type f --exclude=third_party | xargs -L 1 sed -i -r -E 's/^\n+//; s/$/\n/; s/\n+$/\n/g'
```

You can check this PR with:

```bash
git diff --ignore-all-space --ignore-blank-lines "$(git merge-base --fork-point master)"
```

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @mcarilli @ptrblck @leslie-fang-intel @EikanWang @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

**Comment by Skylion007** [comment]
We should add a linter for this like the pre-commit does.

**Comment by Skylion007** [comment]
@XuehaiPan we currently use ufmt. I am thinking about upgrading which linters we use to speed it up a bit though.

**Comment by XuehaiPan** [comment]
@Skylion007 `ufmt` calls `black` and `usort`. We may enable formatting on all files rather than include some of them in `.lintrunner.toml`.

https://github.com/pytorch/pytorch/blob/9171f7d4cdf5d9703f9d22dc1f92bfa0d449a942/.lintrunner.toml#L821-L876

For example, files under `functorch/` are not formatted.

**Comment by XuehaiPan** [comment]
> We should add a linter for this like the pre-commit does.

@Skylion007 I add an initial pre-commit config.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  autofix_prs: true
+  autofix_commit_msg: 'fix: [pre-commit.ci] auto fixes [...]'
+  autoupdate_commit_msg: 'chore(pre-commit): [pre-commit.ci] autoupdate'
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      # - id: trailing-whitespace
```
Comment (XuehaiPan) [inline .pre-commit-config.yaml:14]: Should I enable this in this PR?

**Comment by cpuhrsch** [comment]
Hello @XuehaiPan,

Unfortunately we'll have to close this PR due to the content. Even though this PR isn't small, it still falls under the same reasoning as described in https://github.com/pytorch/pytorch/wiki/Developer-FAQ#my-pr-was-rejected-for-being-too-small--what-gives

Thank you for your contributions!

Christian

**Pull request closed without merging**